src/multiselect.rs

```diff
@@ -131,21 +162,35 @@ impl<T: Clone> PromptInteraction<Vec<T>> for MultiSelect<T> {
     fn render(&mut self, state: &State<Vec<T>>) -> String {
         let theme = THEME.lock().unwrap();
 
-        let line1 = theme.format_header(&state.into(), &self.prompt);
+        let header = theme.format_header(&state.into(), &self.prompt);
+
+        let filter_line = if let Some(input) = self.filter.input() {
+            match state {
+                State::Submit(_) | State::Cancel => "".to_string(),
+                _ => theme.format_input(&state.into(), input),
+            }
+        } else {
+            "".to_string()
+        };
 
-        let mut line2 = String::new();
-        for (i, item) in self.items.iter().enumerate() {
-            line2.push_str(&theme.format_multiselect_item(
+        let mut items_render = String::new();
+        for (i, item) in self.filter.items().iter().map(|i| i.borrow()).enumerate() {
```

Comment:
Then, just replacing `self.items` to `self.filter.items()`. The same in `Select`.